src/gui/editors/PianoRoll.cpp

```diff
@@ -613,55 +622,56 @@ void PianoRoll::setCurrentPattern( Pattern* newPattern )
 		Engine::getSong()->playPattern( NULL );
 	}
 
+	if(m_stepRecorder.isRecording())
+	{
+		m_stepRecorder.stop();
+	}
+
 	// set new data
 	m_pattern = newPattern;
 	m_currentPosition = 0;
 	m_currentNote = NULL;
 	m_startKey = INITIAL_START_KEY;
 
-	if( ! hasValidPattern() )
+	if(hasValidPattern())
```

Comment:
We used to use `if`-`return` pattern to reduce indentation. I think it'll be better if you revert changes from here and add `m_stepRecorder.setCurrentPattern(newPattern);` before this `if` statement if the change does not break anything.